tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/management/TopicControlClusterTest.java

```diff
@@ -50,16 +49,22 @@ public void testClusteredSubscriptionCount() throws Exception {
          Session session2 = conn2.createSession(false, Session.AUTO_ACKNOWLEDGE);
          session2.createDurableSubscriber(topic2, "sub2");
 
-         SimpleString add1 = new SimpleString(topic1.getTopicName());
-         SimpleString add2 = new SimpleString(topic2.getTopicName());
-         AddressControl topicControl1 = ManagementControlHelper.createAddressControl(add1, mBeanServer1);
-         AddressControl topicControl2 = ManagementControlHelper.createAddressControl(add2, mBeanServer2);
+         AddressControl topicControl1 = ManagementControlHelper.createAddressControl(simpleTopicName, mBeanServer1);
+         AddressControl topicControl2 = ManagementControlHelper.createAddressControl(simpleTopicName, mBeanServer2);
+
+         assertTrue("There should be 3 subscriptions on the topic, 2 local and 1 remote.", Wait.waitFor(new Wait.Condition() {
+            @Override
+            public boolean isSatisfied() throws Exception {
+               return topicControl1.getQueueNames().length == 3;
+            }
+         }, 2000));
 
-         assertEquals(2, topicControl1.getQueueNames().length);
-         assertEquals(1, topicControl2.getQueueNames().length);
-      } finally {
-         conn1.close();
-         conn2.close();
+         assertTrue("There should be 3 subscriptions on the topic, 1 local and 2 remote.", Wait.waitFor(new Wait.Condition() {
```

Comment:
We should start using java8 things here.. much cleaner:

For example, expressions
```java
         assertTrue("There should be 3 subscriptions on the topic, 2 local and 1 remote.", Wait.waitFor(() -> topicControl1.getQueueNames().length == 3, 2000));

         assertTrue("There should be 3 subscriptions on the topic, 1 local and 2 remote.", Wait.waitFor(() -> topicControl2.getQueueNames().length == 3, 2000));
```